src/main/java/org/mifos/pheeBillPay/camel/routes/BillerFetchRouteBuilder.java

```diff
@@ -32,28 +35,55 @@ public void configure() {
             logger.debug("Bill Id: " + billId);
             BillerDetails billerDetails = getBillDetails(billId);
             if (billerDetails != null) {
-                exchange.setProperty("billerDetails", billerDetails);
-                exchange.setProperty("billerId", billerDetails.getId());
-                exchange.setProperty("billerName", billerDetails.getBiller());
-                exchange.setProperty("billerType", billerDetails.getBillerCategory());
-                exchange.setProperty("billerAccount", billerDetails.getBillerAccount());
-                exchange.setProperty("billerFetchFailed", false);
+                if(billerDetails.getId().equals(billId)) {
+                    if(billId.equals("004")){
```

Comment:
convert this to ternary operator  expression.
also use an ENUM for 004, to make it more understandable that what is the meaning of 004